tracer/src/Datadog.Trace/Agent/AgentWriter.cs

```diff
@@ -457,12 +457,21 @@ SpanBuffer SwapBuffers()
             // This allows the serialization thread to keep doing its job while a buffer is being flushed
             var buffer = _activeBuffer;
 
-            if (buffer.TryWrite(spans, ref _temporaryBuffer, chunkSamplingPriority))
+            var writeStatus = buffer.TryWrite(spans, ref _temporaryBuffer, chunkSamplingPriority);
+
+            if (writeStatus == SpanBuffer.WriteStatus.Success)
             {
                 // Serialization to the primary buffer succeeded
                 return;
             }
 
+            if (writeStatus == SpanBuffer.WriteStatus.Overflow)
```

Comment:
How does this change interact with partial flushing (if enabled)? 
Could we just enable partial flushing by default instead of make this change?

Reply:
When partial flush is enabled, I expect we will never hit this condition (unless somebody added **a lot** of tags in a span, or tweaked the default threshold). 

I'm in favor of enabling partial flush by default, but it's only compatible with agents 7.26.0+ (https://github.com/DataDog/dd-trace-dotnet/blob/master/tracer/src/Datadog.Trace/Agent/Api.cs#L98) so I don't think the serialization limit is superfluous.